### DISCUSSION TITLE
Any plans on transferring termux.properties to Termux settings?

### DISCUSSION THREAD

**Discussion by TermuxMonet** [termux/termux-app#2984] [Ideas]
Is there any future plans on transferring some of the settings from termux.properties to the settings menu from the app itself?

It's kinda inconvenient that most of the settings from termux are placed on this file. New people wouldn't even know about it's existence, like me, who discovered it in 2021 after 1 year using Termux.

In my opinion, settings being placed on settings menu would be more user-friendly.

**Comment by 2096779623** [comment]
> New people wouldn't even know about it's existence.

Most new people also won't open the settings, they maybe just use the terminal.

**Reply by TermuxMonet** [reply]
I disagree, even users that just begin using the terminal wouldn't find this file, since it is placed in `.termux`, and this folder doesn't appear when doing the `ls` command, only `ls -a`.

The settings are kinda hidden.

**Reply by 2096779623** [reply]
Like `.config` to prevent people from deleting, although it will be regenerate.

**Reply by TermuxMonet** [reply]
And also, it's not explicit that the `.termux` folder contains settings

Someone could just think that it's a cache folder or something like that.
unlike `preferences.d` and `.config`, where users from other distros could recognize it.

**Reply by TermuxMonet** [reply]
But that's only my opinion, this has to be decided by the whole team

**Comment by agnostic-apollo** [comment]
The properties file will remain there and settings won't be moved to shared preferences since its not part of backups and can't update it directly from terminal due to in-memory caching, without an api. However, its planned to make preferences changeable through settings activity, but since `.properties` files also have comments and escapes, so in-place editing would be required to update values. Something like https://github.com/hupfdule/apron could be used. I was thinking that shifting to a yaml config file would provide better structure/scoping, but inplace editing of yaml files would be an issue from the app.

As for file being unknown, new docs have been added for those and will be linked in settings for now. Users are supposed to check wiki anyways, a page already exists for it.

https://wiki.termux.com/wiki/Terminal_Settings

**Reply by Coderx7** [reply]
if theres a mention of termux properties in the settings that would do it imho as well! like a note that says you can access more settings via termux.properties etc etc.

**Reply by agnostic-apollo** [reply]
Yeah, that should work as a temp measure.
